app/controllers/course/leaderboards_controller.rb

```diff
@@ -3,12 +3,17 @@ class Course::LeaderboardsController < Course::ComponentController
   before_action :check_component
   before_action :load_settings
   before_action :add_leaderboard_breadcrumb
+  before_action :check_group_leaderboard_settings, only: [:groups]
```

Comment:
Alternative is to use check_component, but have a case statement for each action.